Source/WebKit/Shared/WebFoundTextRange.h

```diff
@@ -26,17 +26,35 @@
 #pragma once
 
 #include "ArgumentCoders.h"
+#include <variant>
 #include <wtf/HashTraits.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebKit {
 
 struct WebFoundTextRange {
-    uint64_t location { 0 };
-    uint64_t length { 0 };
+    struct DOMData {
+        uint64_t location { 0 };
+        uint64_t length { 0 };
+
+        bool operator==(const DOMData& other) const = default;
+    };
+
+    struct PDFData {
+        uint64_t startPage { 0 };
```

Comment:
I could be wrong, but isn't there like a `PageIndex` type for PDFs that would be better suited here?

Reply:
There is `PDFDocumentLayout::PageIndex`, but I think it might be layering violation to use here.

I'll defer to Abrar.